cryptocurrency-realtime-trend/src/main/java/RedditSource.java

```diff
@@ -0,0 +1,112 @@
+            e.printStackTrace();
+        }
+    }
+
+
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        AbstractProcessor abstractProcessor = new AbstractProcessor() {
+            Set<String> ids = Collections.newSetFromMap(new ConcurrentHashMap<>());
+            RedditClient reddit;
+            public List<String> getSubmissionListForSubreddit(String name) {
+                DefaultPaginator<Submission> paginator = getPaginatorForSubreddit(name);
+                List<Submission> submissions = paginator.accumulateMerged(-1);
+                return analyzeTitleAndContents(submissions);
+            }
+
+            public DefaultPaginator<Submission> getPaginatorForSubreddit(String subredditName) {
+                return reddit
+                        .subreddit(subredditName)
+                        .posts()
+                        .limit(20)
+                        .build();
+
+            }
+
+            public List<String> analyzeTitleAndContents(List<Submission> submissions) {
+                List<String> texts = new ArrayList<String>();
+                for (Submission submission : submissions) {
+                    if (ids.contains(submission.getId())) continue;
+                    ids.add(submission.getId());
+                    texts.add(submission.getTitle() + "." + submission.getSelfText());
+                }
+                ids.clear();
+                return texts;
+            }
+
+            @Override
+            protected void init(Context context) throws Exception {
+                super.init(context);
+
+                // Create our credentials
+                String username = secret.getProperty("username");
+                String password = secret.getProperty("password");
+                String clientId = secret.getProperty("clientId");
+                String clientSecret = secret.getProperty("clientSecret");
+                UserAgent userAgent = new UserAgent("bot", "com.hazelcast.jet", "v1.0", username);
+                Credentials credentials = Credentials.script(username, password, clientId, clientSecret);
+
+                // This is what really sends HTTP requests
+                NetworkAdapter adapter = new OkHttpNetworkAdapter(userAgent);
+
+                // Authenticate and get a RedditClient instance
+                reddit = OAuthHelper.automatic(adapter, credentials);
+                System.out.println("authenticated to reddit ");
+
+            }
+
+            @Override
+            public boolean complete() {
+                try {
+                    Thread.sleep(15000);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                for(String coin:CoinDefs.redditNames){
+                    List<String> strings = getSubmissionListForSubreddit(coin);
+                    for (String string : strings) {
+                        if (!tryEmit(string)) {
```

Comment:
if the emit is rejected, you will lose the list of submissions that were received.